src/components/missionList.js

```diff
@@ -1,20 +1,46 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import '../styles/MissionList.module.css';
+import { useDispatch } from 'react-redux';
+import { joinMission, leaveMission } from 'features/Mission/mission';
 
 function MissionList({
   id, name, description, joined,
 }) {
+  const dispatch = useDispatch();
+  const joinBtn = (event) => {
+    if (event.target.textContent === 'Join Mission') {
+      dispatch(joinMission(event.target.id));
+    }
+    if (event.target.textContent === 'Leave Mission') {
+      dispatch(leaveMission(event.target.id));
+    }
+  };
```

Comment:
## OPTIONAL SUGGESTION

- [ ] I would suggest you check for mission membership using the `joined` variable as opposed to using the button textContent. This will help you prevent logic errors that could occur from typographical errors. 

For example, if you mistakenly misspelled 'Join Mission' your button will not work as it should. check the suggested code below:

```js
const joinBtn = (event) => {
    if (joined) {
      dispatch(leaveMission(event.target.id));
    }else{
      dispatch(joinMission(event.target.id));
    }
};
```

Reply:
Sure, I'll work on it. Thanks